python/ml4ir/base/model/relevance_model.py

```diff
@@ -810,3 +813,44 @@ def calibrate(self, relevance_dataset, logger, logs_dir_local, temperature_init)
                                  logs_dir_local=logs_dir_local,
                                  temperature_init=temperature_init,
                                  file_io=self.file_io)
+
+    def add_temperature_layer(self, temperature: float):
```

Comment:
I am happy you are adding all these. I will repeat something I have insisted in other reviews though. The `base` pipelines and the `base models` are base for a reason: they contain things shared between classification and ranking models. This way we achieve DRY. Functionality that respects this is welcome. If not, it is debt in a sense. 

Now, temperature scaling is not something shared. It only works with classification and [we have a classification model already](https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/applications/classification/model/classification_model.py). This functionality should live there! Is there any reason why you believe this is a better place? 

BTW, see how you are notifying the user with inline comments to be aware for this.

Reply:
yes I agree with you, we do not have calibration with TS for classification models currently. The only reason for  this to be in base class was to have the placeholder for ranking models, too. (we desire to have calibrator for rankers and may have them in future , of course not now)

Reply:
@marjanhs I am happy you agree! Still, it remained in `base`! This function is not a placeholder though. A lot of the assumptions the code does apply only to classification models. Therefore imho it is not generic to be in base. To give you another example, in your imports 
```python
from ml4ir.base.model.calibration.temperature_scaling import temperature_scale,\
    TemperatureScalingLayer
```
It would be saner to have this as: 
```python
from ml4ir.applications.classification.calibration.temperature_scaling import temperature_scale,\
    TemperatureScalingLayer
```

Reply:
that said, it is not the most important issue here perhaps, a lot of nice things happened in the revision below.

Reply:
yes, I am agree with u that calibration works for classification.another reason is that because the ranking model is kind of a classification, we can easily try calibration with a little tweaks with current implementation or implement a totally new calibration for ranking.  That's why I have still kept it in the base.